examples/gpt/schedule.py

```diff
@@ -162,6 +126,21 @@ def pattern(x: torch.Tensor) -> torch.Tensor:
             sub_sch["module.FusedQKV_0.fused_linear"].shard("bias", axis=0)
             sub_sch["module.out_proj"].shard("weight", axis=1)
 
+            # Attention mask is broadcasted from (B, 1, 1, S) to (B, H, S, S),
+            # where H is sharded.
+            ops = sub_sch["module"].find_node(
+                lambda node: node.op == "call_method" and node.target == "repeat"
+            )
+            assert len(ops) == 1
+
+            def new_repeat(tensor, *args):
+                # (B, 1, 1, S) -> (B, H, S, S)
+                assert len(args) == 4
+                out = tensor.repeat(args[0], args[1] // sch.world_size, *args[2:])
```

Comment:
this looks like a big memory overhead, especially for sequence length 8192 ... and it will cancel out the memory efficiency of flash attention kernel

Reply:
Per offline discussion with @jfc4050, now we use the following strategies:
1. For cutlass kenrel in xFormers, it has to take the bias in bhqk shape, but it could be expended logically from b11k, so we use `tensor.expand` to avoid actual memory copy.
2. For triton kernel in flash-attention, we directly use b11k.